Description: Prevent overflow of a CARD32 variable millis by doing bounds checking.
 This is a patch currently in discussion at fdo #27134, so this patch can be dropped in favor of whatever upstream decides to include.
Author: Bryce Harrington <bryce@ubuntu.com>
Bug-Ubuntu: https://launchpad.net/bugs/519049

Index: xorg-server-1.17.2/Xext/security.c
===================================================================
--- xorg-server-1.17.2.orig/Xext/security.c
+++ xorg-server-1.17.2/Xext/security.c
@@ -261,9 +261,9 @@ SecurityComputeAuthorizationTimeout(Secu
     /* maxSecs is the number of full seconds that can be expressed in
      * 32 bits worth of milliseconds
      */
-    CARD32 maxSecs = (CARD32) (~0) / (CARD32) MILLI_PER_SECOND;
+    CARD32 maxSecs = (CARD32) (MAXINT) / (CARD32) MILLI_PER_SECOND;
 
-    if (seconds > maxSecs) {    /* only come here if we want to wait more than 49 days */
+    if (seconds > maxSecs) {    /* only come here if we want to wait more than 24 days */
         pAuth->secondsRemaining = seconds - maxSecs;
         return maxSecs * MILLI_PER_SECOND;
     }
@@ -297,8 +297,6 @@ SecurityAuthorizationExpired(OsTimerPtr
 {
     SecurityAuthorizationPtr pAuth = (SecurityAuthorizationPtr) pval;
 
-    assert(pAuth->timer == timer);
-
     if (pAuth->secondsRemaining) {
         return SecurityComputeAuthorizationTimeout(pAuth,
                                                    pAuth->secondsRemaining);
